apps/styleguide/src/tables/RowEditModal.vue

```diff
@@ -108,6 +116,9 @@ export default {
     saveDraft() {
       this.executeCommand(true);
     },
+    saveEditDraft() {
+      this.executeCommand(true);
```

Comment:
executeCommand is a pretty generic function, based on the code I would say it queries the backend with graphQL, maybe rename to make it more clear what command is executed?

Reply:
it can execute either command 'insert' or 'update' depending on whether we show a form for 'create' (=insert) or for 'update'. What would you recommend?

Reply:
if it's only those two options, you could make it an upsert

https://dgraph.io/docs/graphql/mutations/upsert/

Reply:
that is not the same. Currently it is either insert or update. That means the validation is different, because 'insert' will fail if record already exists. And that is desired if you push '+'

Reply:
would it greatly help your understanding we would pass the 'update' and 'insert' as a parameter?

Reply:
actually, the only difference I can spot in executeCommand is

```
      if (isDraft) {
        this.value["mg_draft"] = true;
      } else {
        this.value["mg_draft"] = false;
      }
```

this.value is a data property, which you mutate directly. So you can do this before executeCommand, removing the need of the boolean at all. Then you can rename executeCommand to save, persistToDatabase or something else that covers more what it does.

SaveEditdraft, does exactly the same as savedraft, so you can use both times savedraft

Sidenote, I don't recommend using a data property called value, as it's very generic and confusing with using v-model on a custom component (it uses 'value' as a prop)

So code could look like:

```
 saveDraft() { 
       this.record["mg_draft"] = true;
      this.save();
    },
```

even better is to extract the save logic outside of the component and call it like this:
```
this.save(this.record)
```